Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow.lean

```diff
@@ -0,0 +1,339 @@
+    exact ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const <| Or.inl <| by aesop
+  have h₂ : ContinuousOn (fun z : ℝ≥0 => z ^ (x : ℝ)) (spectrum ℝ≥0 a) := by
+    intro z hz
+    exact ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const <| Or.inl <| by aesop
+  rw [← cfc_comp _ _ a ha₂ h₁ h₂]
+  refine cfc_congr fun _ _ => ?_
+  simp [NNReal.rpow_mul]
+
+lemma rpow_rpow_of_exponent_nonneg {a : A} {x y : ℝ} (hx : 0 ≤ x) (hy : 0 ≤ y)
+    (ha₂ : 0 ≤ a := by cfc_tac) : (a ^ x) ^ y = a ^ (x * y) := by
+  simp only [rpow_def]
+  have h₁ : ContinuousOn (fun z : ℝ≥0 => z ^ (y : ℝ))
+      ((fun z : ℝ≥0 => z ^ (x : ℝ)) '' spectrum ℝ≥0 a) :=
+    fun _ _ => ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const (Or.inr hy)
+  have h₂ : ContinuousOn (fun z : ℝ≥0 => z ^ (x : ℝ)) (spectrum ℝ≥0 a) :=
+    fun _ _ => ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const (Or.inr hx)
+  rw [← cfc_comp _ _ a ha₂ h₁ h₂]
+  refine cfc_congr fun _ _ => ?_
+  simp [NNReal.rpow_mul]
+
+lemma rpow_one {a : A} (ha : 0 ≤ a := by cfc_tac) : a ^ (1 : ℝ) = a := by
+  simp only [rpow_def, NNReal.coe_one, NNReal.rpow_eq_pow, NNReal.rpow_one]
+  change cfc (id : ℝ≥0 → ℝ≥0) a = a
+  rw [cfc_id ℝ≥0 a]
+
+@[simp]
+lemma one_rpow {x : ℝ} : (1 : A) ^ x = (1 : A) := by simp [rpow_def]
+
+lemma rpow_zero {a : A} (ha : 0 ≤ a := by cfc_tac) : a ^ (0 : ℝ) = 1 := by
+  simp [rpow_def, cfc_const_one ℝ≥0 a]
+
+lemma zero_rpow {x : ℝ} (hx : x ≠ 0) : rpow (0 : A) x = 0 := by simp [rpow, NNReal.zero_rpow hx]
+
+section unital_vs_nonunital
+
+variable [NonUnitalContinuousFunctionalCalculus ℝ≥0 (fun (a : A) => 0 ≤ a)]
+  [UniqueNonUnitalContinuousFunctionalCalculus ℝ≥0 A]
+
+lemma rpowₙ_eq_rpow {a : A} {x : ℝ≥0} (hx : 0 < x) : rpowₙ a x = a ^ (x : ℝ) := by
+  have hzero : NNReal.rpow 0 x = 0 := by
+    have : x ≠ 0 := ne_of_gt hx
+    simp [this]
+  have hcont : ContinuousOn (fun z => NNReal.rpow z x) (quasispectrum ℝ≥0 a) :=
+    fun _ _ => ContinuousAt.continuousWithinAt
+      <| NNReal.continuousAt_rpow_const (Or.inr NNReal.zero_le_coe)
+  rw [rpowₙ, rpow_def, cfcₙ_eq_cfc hcont hzero]
+  rfl
+
+lemma sqrt_eq_rpow {a : A} : sqrt a = a ^ (1 / 2 : ℝ) := by
+  have : a ^ (1 / 2 : ℝ) = a ^ ((1 / 2 : ℝ≥0) : ℝ) := rfl
+  rw [this, ← rpowₙ_eq_rpow (by norm_num), sqrt_eq_rpowₙ]
+
+lemma sqrt_eq_cfc {a : A} : sqrt a = cfc NNReal.sqrt a := by
+  rw [sqrt, cfcₙ_eq_cfc]
+
+lemma sqrt_sq {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt (a ^ 2) = a := by
+  rw [pow_two, sqrt_mul_self]
+
+lemma sq_sqrt {a : A} (ha : 0 ≤ a := by cfc_tac) : (sqrt a) ^ 2 = a := by
+  rw [pow_two, sqrt_mul_sqrt_self]
+
+@[simp]
+lemma sqrt_algebraMap {r : ℝ≥0} : sqrt (algebraMap ℝ≥0 A r) = algebraMap ℝ≥0 A (NNReal.sqrt r) := by
+  rw [sqrt_eq_cfc, cfc_algebraMap]
+
+@[simp]
+lemma sqrt_one : sqrt (1 : A) = 1 := by simp [sqrt_eq_cfc]
+
+-- TODO: relate to a strict positivity condition
+lemma sqrt_rpow_of_spectrum_pos {a : A} {x : ℝ} (h : ∀ z ∈ spectrum ℝ≥0 a, 0 < z)
```

Comment:
I didn't say this before, but could have: since this is the `ℝ≥0`-spectrum, not including zero is equivalent to being positive.
```suggestion
lemma sqrt_rpow_of_spectrum_pos {a : A} {x : ℝ} (h : 0 ∉ spectrum ℝ≥0 a)
```

Reply:
Fair enough.